gptresearcher/context/research_context.py

```diff
@@ -5,44 +5,40 @@
 import json
 import hashlib
 
-from actions.web_search import web_search
-from actions.web_scrape import async_browse
-from processing.text import \
+from gptresearcher.actions.web_search import web_search
+from gptresearcher.actions.web_scrape import async_browse
+from gptresearcher.context.llm_utils import choose_agent
+from gptresearcher.processing.text import \
     write_to_file, \
     create_message, \
     create_chat_completion, \
     read_txt_files, \
     write_md_to_pdf
 from config import Config
-from agent import prompts
+from gptresearcher.context import prompts
 import os
-import string
 
 
-CFG = Config()
-
-
-class ResearchAgent:
-    def __init__(self, question, agent, agent_role_prompt, websocket=None):
-        """ Initializes the research assistant with the given question.
-        Args: question (str): The question to research
-        Returns: None
-        """
-
+class ResearchContext:
```

Comment:
@gregdrizz as discussed it's better consider it as "RetrieverAgent" than Context. Makes more sense. That way in theory developers or us can initiate multiple RetrieverAgents based on various sources/use cases/domains, etc. And again, instead of passing a researcher object, not sure why not to pass the agent_role, agent, etc as originally planned? It would make this object be more abstract and customizable for various sub research agents.